flw/src/main/java/org/motechproject/nms/flw/service/impl/ServiceUsageServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package org.motechproject.nms.flw.service.impl;
+
+import org.joda.time.DateTime;
+import org.motechproject.mds.query.QueryExecution;
+import org.motechproject.mds.util.InstanceSecurityRestriction;
+import org.motechproject.nms.flw.domain.FrontLineWorker;
+import org.motechproject.nms.flw.domain.ServiceUsage;
+import org.motechproject.nms.flw.repository.ServiceUsageDataService;
+import org.motechproject.nms.flw.service.ServiceUsageService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.jdo.Query;
+import java.util.List;
+
+/**
+ * Simple implementation of the {@link org.motechproject.nms.flw.service.FrontLineWorkerService} interface.
+ */
+@Service("serviceUsageService")
+public class ServiceUsageServiceImpl implements ServiceUsageService {
+    private ServiceUsageDataService serviceUsageDataService;
+
+    @Autowired
+    public ServiceUsageServiceImpl(ServiceUsageDataService serviceUsageDataService) {
+        this.serviceUsageDataService = serviceUsageDataService;
+    }
+
+    @Override
+    public void add(ServiceUsage record) {
+        serviceUsageDataService.create(record);
+    }
+
+    @Override
+    public ServiceUsage getCurrentMonthlyUsageForFLWAndService(final FrontLineWorker frontLineWorker, final org.motechproject.nms.flw.domain.Service service) {
+        ServiceUsage serviceUsage = new ServiceUsage(frontLineWorker, service, 0, 0, 0, DateTime.now());
+
+        QueryExecution<List<ServiceUsage>> queryExecution = new QueryExecution<List<ServiceUsage>>() {
```

Comment:
Adding ```@SuppressWarnings("unchecked")``` will silence the obnoxious "unchecked or unsafe operations." compile warning.